caikit_computer_vision/modules/segmentation/image_processing.py

```diff
@@ -0,0 +1,286 @@
+        detectron2_out=coco_instances,
+        inf_coco=coco_json,
+        score_threshold=threshold,
+        save_coco="",
+    )
+    coco_json = inf_coco.dataset
+    return coco_json
+
+
+def _convert_results_dict_to_instances(results: dict) -> "Instances":
+    """Helper function to convert output of model to Instances.
+
+    Args:
+        results (dict): Raw COCO Dict from Detectron2.
+
+    Returns:
+        Instances: Instances object.
+    """
+    boxes: Boxes = Boxes(results["BOXES"])
+    classes: torch.Tensor = torch.as_tensor(results["CLASSES"])
+    masks: torch.Tensor = torch.as_tensor(results["MASKS"])
+    scores: torch.Tensor = torch.as_tensor(results["SCORES"])
+    image_size = results["IMAGES_SIZE"]
+    instances = Instances(
+        image_size=image_size,
+        pred_boxes=boxes,
+        scores=scores,
+        pred_classes=classes,
+        pred_masks=masks,
+    )
+    return instances
+
+
+def _assemble_images_coco(
+    file_name: str,
+    height: int,
+    width: int,
+    image_id: int = 1,
+) -> "COCO":
+    """Assembles coco dictionary into file and returns COCO object.
+
+    Args:
+        file_name (str): File name
+        height (int): Height of image
+        width (int): Width of image
+        image_id (int, optional): Optional image id. Defaults to 1.
+
+    Returns:
+        COCO: COCO Object
+    """
+    coco_json = dict(
+        images=[
+            {
+                "id": image_id,
+                "file_name": file_name,
+                "height": height,
+                "width": width,
+                "license": 1,
+            }
+        ],
+    )
+    with NamedTemporaryFile(suffix=".json") as tmp_file:
+        with open(tmp_file.name, "w") as f_temp:
+            json.dump(coco_json, f_temp)
+        coco = COCO(tmp_file.name)
+    return coco
+
+
+def detectron2_to_coco(
+    detectron2_out: Union[dict, str],
+    inf_coco: Union[Type[COCO], str],
+    score_threshold: float = 0.2,
+    save_coco: str = "",
+) -> Type[COCO]:
+    """convert detectron2 inference output to coco.
+
+    Args:
+      detectron2_out  : path to inference output .json file or dictionary with detectron2 inference output
+      inf_coco        : inference coco file, required to contain 'images' and 'categories' keys
+      score_threshold : minimum score threshold to keep an annotation
+      save_coco       : if a save_coco is provided, then the coco annotations are saved to JSON
+                        with the file name 'coco_results_poly.json' under the path provided in save_coco
+
+    Returns:
+      coco_dict : coco formatted dict of annotations
+    """
+
+    if isinstance(detectron2_out, str):
+        with open(detectron2_out, "r") as f_out:
+            detectron2_out = json.load(f_out)
+
+    if isinstance(inf_coco, str):
+        inf_coco = COCO(inf_coco)
+    else:
+        error.type_check("<CCV61312083E>", COCO, inf_coco=inf_coco)
+
+    error.value_check(
+        "<CCV61312083E>",
```

Comment:
Ah sorry to be clear - the point of these is so that each code has a unique UUID, so the numbers for each type/value check etc should be different. This way, if it comes out through the runtime, it's easy to locate the exact line of code that produced the error message!